dotCMS/src/main/java/com/dotcms/rest/MapToContentletPopulator.java

```diff
@@ -332,14 +336,47 @@ private void processSeveralHostsOrFolders(final Contentlet contentlet,
         }
     } // processSeveralHostsOrFolders.
 
+    /**
+     * @deprecated Use {@link MapToContentletPopulator#fetchCategories(Contentlet, User, boolean)} instead
+     * @param contentlet
+     * @param user
+     * @param respectFrontendRoles
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Deprecated
     @CloseDBIfOpened
     public List<Category> getCategories (final Contentlet contentlet, final User user,
                                          final boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
 
         DotPreconditions.checkNotNull(contentlet, IllegalArgumentException.class, "Invalid Contentlet");
         DotPreconditions.checkNotNull(contentlet.getContentType(), IllegalArgumentException.class, "Invalid Content Type");
 
-        return categories(contentlet, user, respectFrontendRoles);
+        return internalFetchCategories(contentlet, user, respectFrontendRoles).orElseGet(
+                ImmutableList::of);
+    }
+
+    /**
+     * The List of Categories generated by this method is what will be set directly into the checkin method
+     * Meaning that if this method generates an empty list, The categories in the contentlet will be wiped-out. An empty Optional means nothing should be sent down into the checkin pipeline
+     * @param contentlet
+     * @param user
+     * @param respectFrontendRoles
+     * @return Empty Optional means no categories were found if something comes back within the Optional that's what must be applied there.
+     * If an empty Collection of categories is returned that means we need to wipe out the categories from the contentlet
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @CloseDBIfOpened
```

Comment:
Why a CloseDBIfOpened is required here?

Reply:
I'm deprecating getCategories in favor of `fetchCategories` and.. since they both do they same I simply repeated the pattern. Though if you think about it we now have a method that loads the existing categories which makes use of the db